test/unit/puppetlabs/services/master/master_service_test.clj

```diff
@@ -70,3 +69,52 @@
                   (test-path! "cert_inventory" "target/master-service-test/inventory.txt"))))))
         (finally
           (fs/delete-dir test-dir))))))
+
+(deftest version-check-test
+  (testing "master calls into the dujour version check library using the correct values"
+    ; This atom will store the parameters passed to the version-check-test-fn, which allows us to keep the
+    ; assertions about their values inside the version-check-test and will also ensure failures will appear if
+    ; the master stops calling the check-for-updates! function
+    (let [version-check-params  (atom {})
+          version-check-test-fn (fn [request-values update-server-url]
+                                  (swap! version-check-params #(assoc % :request-values request-values
+                                                                        :update-server-url update-server-url)))
+          test-dir (doto "target/master-service-test" fs/mkdir)]
+      (try
+        (with-redefs
+          [version-check/check-for-updates! version-check-test-fn
+           ; Redefine the CA functions to ensure that the master does not create
+           ; any SSL files
+           ca/retrieve-ca-cert! (fn [_ _])
+           ca/retrieve-ca-crl! (fn [_ _])
+           ca/initialize-master-ssl! (fn [_ _ _])]
```

Comment:
Could you use the disabled CA service instead of the actual CA service + with-redefs?